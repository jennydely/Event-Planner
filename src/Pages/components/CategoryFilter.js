import React from 'react'
import SelectCategory from './common/SelectCategory'

export default function CategoryFilter({ onSelectFilter }) {
  const categories = [
    'category',
    'all',
    'metal',
    'medieval',
    'holiday',
    'other',
  ]

  return (
    <SelectCategory
      data-testid="my-select-component"
      id="category"
      placeholder="category"
      name={categories}
      options={categories}
      onSelectFilter={onSelectFilter}
    />
  )
}
