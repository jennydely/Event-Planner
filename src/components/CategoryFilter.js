import React from 'react'
import styled from 'styled-components/macro'
import SelectCategory from '../common/SelectCategory'

export default function CategoryFilter({ onSelectFilter }) {
  const categories = ['all', 'metal', 'medieval', 'holiday', 'other']

  return (
    <FilterContainer data-testid="my-select-component">
      <SelectCategory
        id="category"
        placeholder="all"
        name={categories}
        options={categories}
        onSelectFilter={onSelectFilter}
      />
    </FilterContainer>
  )
}

const FilterContainer = styled.div`
  margin: 5px;
`
