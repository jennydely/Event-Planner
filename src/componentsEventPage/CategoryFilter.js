import React from 'react'
import styled from 'styled-components/macro'
import SelectCategory from '../common/SelectCategory'


export default function CategoryFilter({ onSelectFilter }) {
 
  return (
    <FilterContainer>
      <SelectCategory id="category" name={["metal", "medieval", "sand", "other"]} 
        options={["metal", "medieval", "sand", "other"]}  onSelectFilter={onSelectFilter}/>
      <CategoryButton name="metal" onClick={() => onSelectFilter('metal')}>metal</CategoryButton>
      <CategoryButton name="medieval" onClick={() => onSelectFilter('medieval')}>medieval</CategoryButton>
      <CategoryButton name="sand" onClick={() => onSelectFilter('sand')}>sand</CategoryButton>
      <CategoryButton name="other" onClick={() => onSelectFilter('other')}>other</CategoryButton>
    </FilterContainer>
  )
}

const FilterContainer = styled.div`
margin 5px;
backgroundcolor: grey;
`

const CategoryButton = styled.button`
background: ${({ name }) => name === 'sand' ? 'rgba(248,149,17,0.46)' : (name === 'metal' ? 'rgba(49,42,42,0.75)' : (name === 'medieval' ? 'rgba(67,40,24,0.70)' : (name === 'other' ? 'rgba(153,88,42,0.70)' : 'rgb(96,99,104)')))};
` 