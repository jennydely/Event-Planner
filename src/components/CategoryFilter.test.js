import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import CategoryFilter from './CategoryFilter'
import 'jest-styled-components'

describe('CategoryFilter', () => {
  const onSelectFilter = jest.fn()

  it('displays all buttons to filter and test the filter function', () => {
    const { getByText } = render(
      <CategoryFilter onSelectFilter={onSelectFilter} />
    )
    expect(getByText('all')).toBeInTheDocument()
    expect(getByText('medieval')).toBeInTheDocument()
    expect(getByText('metal')).toBeInTheDocument()
    expect(getByText('holiday')).toBeInTheDocument()
    expect(getByText('other')).toBeInTheDocument()
    fireEvent.click(getByText('medieval'))
    expect(onSelectFilter).toHaveBeenCalledWith('medieval')
  })
  it('renders correctly', () => {
    const tree = render(<CategoryFilter onSelectFilter={onSelectFilter} />)
    expect(tree).toMatchSnapshot()
  })
})
