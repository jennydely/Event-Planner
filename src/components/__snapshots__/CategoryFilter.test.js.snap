// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  color: var(--lightyellow-90);
  background: var(--lightgrey-main);
}

.c2 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 112.5%;
  color: var(--lightyellow-90);
  background: var(--lightgrey-main);
}

.c3 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 112.5%;
  color: var(--lightyellow-90);
  background: var(--darkgrey-75);
}

.c4 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 112.5%;
  color: var(--lightyellow-90);
  background: var(--darkbrown-70);
}

.c5 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 112.5%;
  color: var(--lightyellow-90);
  background: var(--blue-70);
}

.c6 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 112.5%;
  color: var(--lightyellow-90);
  background: var(--lightbrown-70);
}

.c0 {
  margin: 5px;
}

<body>
    <div>
      <div
        class="c0"
      >
        <select
          class="c1"
          id="category"
          name="all,metal,medieval,holiday,other"
        >
          <option
            class="c2"
          >
            all
          </option>
          <option
            class="c3"
          >
            metal
          </option>
          <option
            class="c4"
          >
            medieval
          </option>
          <option
            class="c5"
          >
            holiday
          </option>
          <option
            class="c6"
          >
            other
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CategoryFilter__FilterContainer-rogkyz-0 fmiTmv"
    >
      <select
        class="SelectCategory__SelectStyled-sc-1n16935-0 kPEDeF"
        id="category"
        name="all,metal,medieval,holiday,other"
      >
        <option
          class="SelectCategory__OptionStyled-sc-1n16935-1 kGYOSt"
        >
          all
        </option>
        <option
          class="SelectCategory__OptionStyled-sc-1n16935-1 JZlKi"
        >
          metal
        </option>
        <option
          class="SelectCategory__OptionStyled-sc-1n16935-1 eTyMJM"
        >
          medieval
        </option>
        <option
          class="SelectCategory__OptionStyled-sc-1n16935-1 kSaATc"
        >
          holiday
        </option>
        <option
          class="SelectCategory__OptionStyled-sc-1n16935-1 gxfQXm"
        >
          other
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
